pkg/api/handlers/slackevent.go

```diff
@@ -0,0 +1,148 @@
+package handlers
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/gptscript-ai/go-gptscript"
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	"github.com/obot-platform/obot/pkg/invoke"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+)
+
+type SlackEventHandler struct {
+	invoker   *invoke.Invoker
+	gptscript *gptscript.GPTScript
+}
+
+func NewSlackEventHandler(invoker *invoke.Invoker, gptscript *gptscript.GPTScript) *SlackEventHandler {
+	return &SlackEventHandler{invoker: invoker, gptscript: gptscript}
+}
+
+type SlackEvent struct {
```

Comment:
in their docs, they make a big deal about verify thing token

https://api.slack.com/apis/events-api
```

The shared-private callback token that authenticates this callback to the application as having come from Slack. Match this against what you were given when the subscription was created. If it does not match, do not process the event and discard it. Example: JhjZd2rVax7ZwH7jRYyWjbDl
```

Should we be checking that as well?

Reply:
Yeah this is deprecrated in favor of using signing secret to verify the payload

<img width="683" alt="image" src="https://github.com/user-attachments/assets/e9b20321-297c-4922-802e-6e5f04be2a42" />

Reply:
Yeah this is deprecated. Using the sign secret is in favor of.

> This deprecated Verification Token can still be used to verify that requests come from Slack, but we strongly recommend using the above, more secure, signing secret instead.